File: src/test/scala/com/github/kayjamlang/tests/expressions/AssertNullExpressionTest.scala
package com.github.kayjamlang.tests.expressions

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.expressions.{AssertNullExpression, ValueExpression}
import org.junit.Assert.{assertNotNull, assertSame}
import org.junit.{BeforeClass, Test}

object AssertNullExpressionTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("123!"))
  }
}

class AssertNullExpressionTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = AssertNullExpressionTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[AssertNullExpression], expression getClass)
    val assertNull = expression.asInstanceOf[AssertNullExpression]
    assertSame(classOf[ValueExpression], assertNull.expression getClass)
  }
}

File: src/test/java/com/github/kayjamlang/tests/expressions/AssertNullExpressionTest.java
package com.github.kayjamlang.tests.expressions;

import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.expressions.AssertNullExpression;
import com.github.kayjamlang.core.expressions.ValueExpression;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertSame;

public class AssertNullExpressionTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("123!"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(AssertNullExpression.class, expression.getClass());

        AssertNullExpression assertNull = (AssertNullExpression) expression;
        assertSame(ValueExpression.class, assertNull.expression.getClass());
    }
}